hs_composite_resource/models.py

```diff
@@ -217,16 +479,51 @@ def update_fileset_folder():
                 fs.folder = new_folder + fs.folder[len(old_folder):]
                 fs.save()
 
+        def update_model_program_folder():
```

Comment:
I'm just going to note generally in the composite resource, anytime the composite resource is checking logical file types, it's a signal that the logic should be pushed to the logical file types.  I can't think of a situation where the composite resource should be handling different logical files differently based on the types.  That should be baked into the logical file design.  

I know there were already quite a few checks for filesets in the composite resource... but I think even the logic around those checks could/should be pushed into the logical file.